lib/SIL/SILFunctionBuilder.cpp

```diff
@@ -67,33 +65,24 @@ void SILFunctionBuilder::addFunctionAttributes(SILFunction *F,
   if (Attrs.hasAttribute<SILGenNameAttr>() || Attrs.hasAttribute<CDeclAttr>())
     F->setHasCReferences(true);
 
+  // NOTE: Validate `@differentiable` attributes by calling
+  // `getParameterIndices`. This is important for:
+  // - Skiping invalid `@differentiable` attributes in non-primary files.
+  // - Preventing duplicate SIL `[differentiable]` attribute generation for
+  //   `@differentiable` attributes on `AbstractStorageDecl` declarations.
+  //   Such attributes are deleted and recreated on the getter `AccessorDecl`
+  //   of the `AbstractStorageDecl`.
+  // NOTE(TF-988): Consider create SIL differentiability witnesses from
+  // `@differentiable` and `@derivative` attributes here instead of in
+  // `SILGenModule::postEmitFunction`.
+  for (auto *A : Attrs.getAttributes<DifferentiableAttr>())
+    (void)A->getParameterIndices();
+
   // Propagate @_dynamicReplacement(for:).
   if (constant.isNull())
     return;
   auto *decl = constant.getDecl();
 
-  // SWIFT_ENABLE_TENSORFLOW
-  // Propagate @differentiable attributes.
-  // Don't propagate @differentiable to:
-  // - Non-getter accessors (setters, modifiers, etc).
-  // - Default argument generator functions.
-  // - Thunks. Those are currently handled in SILGenThunk.cpp.
-  if ((!isa<AccessorDecl>(decl) || cast<AccessorDecl>(decl)->isGetter()) &&
```

Comment:
Note: reviving this code may be necessary for TF-988: creating SIL differentiability witnesses in `SILFunctionBuilder` utilities instead of `SILGenModule::postEmitFunction`.

For simplicity, I decided to remove the code for now - the diff exists in the history if we need it. To avoid the TF-1057 crash, we could create a new fileprivate helper just for processing `@differentiable` attributes instead of reusing `SILFunctionBuilder::addFunctionAttributes`.